### ISSUE TITLE
Segfault from Linkage::Criteria::expandToTarget()

### ISSUE THREAD

**Issue by jslee02** [dartsim/dart#489]
[Linkage.cpp Ln:364](https://github.com/dartsim/dart/blob/master/dart/dynamics/Linkage.cpp#L365)
```cpp
  // If we have successfully reached the target, expand from there
  if(newBns.back() == _target.mNode.lock())
    expansionPolicy(_target.mNode.lock(), _target.mPolicy, newBns);
```

Segfault is raised at `newBns.back()` when `newBns` is empty. According to [cppreference](http://en.cppreference.com/w/cpp/container/vector/back), calling back on an empty container is undefined. I observed that this segfault is happened on Windows but not Ubuntu or Mac.

**Comment by mxgrey** [comment]
This is actually a really easy fix. This is just an unhandled edge case where the linkage wasn't able to expand towards the target whatsoever. The current code incorrectly assumes that it was able to take at least one step towards the target. We just need to change the if-statement to ``if(newBns.size() > 0 && newBns.back() == _target.mNode.lock())``. I'll make a pull request with the fix.

**Comment by jslee02** [comment]
Thanks! I confirmed that it fixes the segfault. I just made a comment on #490 about minor code preference.

**Comment by jslee02** [comment]
Fixed by #490, #491, and #492.
